Creating the external foundry documentation structure and first round of documentation for RFD3

rclune opened the pull request:
- Created a workflow to automatically build docs on PR or Push to production
- Created the infrastructure for foundry documentation. `foundry/docs` is where the documentation needs to be built, it has symlinks to the documentation folders for each of the models. The individual models have docs folders that can be populated with model-specific documentation. (Contribution guide in future PR.)
- Created a PPI design tutorial with Rafi
- Updated input.md to more thoroughly discuss the different CLI and input options
- Created an `inference basics` document to help get people started, understand what they will get out of running RFD3
- renamed input files to remove typo (croppped -> cropped)

rclune committed:
Merge branch 'production' into rac_docs

rclune committed:
Starting to put together the foundry and RFD3 external documentation

Set up the foundation for Sphinx to be able to build the external docs, first draft of a ppi_design_tutorial has been completed.

rclune committed:
Add RFdiffusion3 documentation and update toctree

Added initial documentation for RFdiffusion3 under models/rfd3/docs/index.rst and linked it in the main docs toctree. Updated toctree maxdepth for better navigation and added a symlink for the rfd3 model documentation.

rclune committed:
Update and expand RFdiffusion3 documentation

Added introductory and reference documentation files, improved the index structure with general information and examples, and enhanced the PPI design tutorial with additional notes, figures, and clarifications. Also fixed a typo in a PDB filename.

rclune committed:
Update RFD3 documentation and tutorial content

Expanded the Sphinx static path to include RFD3 assets and made minor formatting and clarity improvements in the main and RFD3-specific documentation. The PPI design tutorial was revised for clarity, improved step-by-step instructions, and better separation of setup and execution steps.

rclune committed:
Update RFD3 documentation and tutorials

Added source_suffix to Sphinx conf.py for Markdown support. Updated index.rst to include new documentation sections. Expanded intro_inference_calculations.md with detailed instructions on inference input formats, job configuration, and output files. Improved input.md formatting for appendices and FAQs. Revised ppi_design_tutorial.md for clarity, added details on settings, and expanded explanations for hotspots and batch inference.

rclune committed:
Update RFD3 docs: clarify input specs and file formats

Expanded and clarified the documentation for RFdiffusion3 input specifications, including more detailed explanations of the 'contig' string, input file types, and example YAML/JSON formats. Improved the intro to inference calculations to better explain the structure and usage of settings files, and updated descriptions for job configuration and output files. Added a placeholder for configuration options documentation.

rclune committed:
Update RFdiffusion3 input documentation and examples

Expanded and clarified the documentation for RFdiffusion3 input specification, including detailed explanations of CLI arguments, InputSpecification fields, the InputSelection mini-language, contig string formatting, and advanced options such as partial diffusion and CIF parser arguments. Added more examples, debugging recommendations, and an updated FAQ. Also updated the output file naming explanation for clarity. Removed the obsolete configuration_options.md file.

Note that images are still not being rendered correctly in many of the md files. Fix will be in future commit.

rclune committed:
PPI tutorial and RFD3 docs update

Created output files for PPI tutorial and listed their locations. Made edits to files to add labels to sections to remove sphinx warnings.

rclune committed:
Merge branch 'production' into rac_docs

Ubiquinone-dot commented on docs/source/conf.py~:
```diff
@@ -0,0 +1,28 @@
+# https://www.sphinx-doc.org/en/master/usage/configuration.html
+
+# -- Project information -----------------------------------------------------
+# https://www.sphinx-doc.org/en/master/usage/configuration.html#project-information
```
Not super faimiliar with docs building with sphinx, but does this file need to end with ~?

rclune replied:
Ah sorry, this is an emacs file thing, didn't mean to get this caught in the commit. Thank you!

Ubiquinone-dot commented on docs/source/models/rfd3:
```diff
@@ -0,0 +1 @@
+/Users/rclune/foundry/models/rfd3/docs
```
Looks like this is a stray path

rclune replied:
Thanks for catching this! I the symlink is necessary for sphinx to build the docs correctly, but I made it a relative path.

Ubiquinone-dot commented on models/rfd3/docs/index.rst:
```diff
@@ -0,0 +1,33 @@
+
+RFdiffusion3 is a powerful protein design tool that operate on the atomic level to 
+study ligand-protein interactions, create nucleic acid-protein interfaces, and 
+design enzymes with specific catalytic activities. It is designed to be highly flexible and 
```
Nit: `with specific catalytic activities.` -> `*de novo*`

Ubiquinone-dot commented on models/rfd3/docs/input.md:
```diff
@@ -142,45 +243,78 @@ In the following example, RFD3 will noise out by 15 angstroms and constrain atom
 
-Let us know if you have any additional questions, we'd be happy to answer them!
+
+Let us know if you have any additional questions, we'd be happy to answer them either in our [Slack channel](https://join.slack.com/t/proteinmodelfoundry/shared_invite/zt-3kpwru8c6-nrmTW6LNHnSE7h16GNnfLA) or in a GitHub discussion. 
```
Nice!!

Ubiquinone-dot commented on models/rfd3/docs/intro_inference_calculations.md:
````diff
@@ -0,0 +1,106 @@
+Some settings will be one line while others will require defining a dictionary, leading to another set of braces: 
+```json
+{
+    "inference_calculation_1":{
````
Since these don't actually relate to rfd3 I think it comes across as a bit misleading; we can put a note to refer people to yaml or json syntaxes but I would wager most people are familiar with it

rclune replied:
Removed those sections and pointed people towards the provided examples to see the json/yaml file format.

Ubiquinone-dot reviewed (changes requested):
Few minor nits but mostly lgtm

rclune committed:
Delete docs/source/conf.py~

This is an auto-save file from emacs - it does not need to be in the repo.

rclune committed:
Delete docs/source/index.rst~

This is an auto-saved copy of index.rst,  it does not need to be in the repo

rclune committed:
Adding missing images and fixing docs symlink

rclune committed:
Fix grammatical error in RFdiffusion3 documentation

Fixed typo, clarified enzyme design language.

rclune committed:
Fix typos and enhance clarity in inference docs

Corrected typos and improved clarity in the documentation regarding inference settings and file formats.
